### PR TITLE
[WFLY-15547]: Move WildFly Preview to a native Jakarta namespace vari…

### PR DESCRIPTION
**Pull request by ehsavoie**
…ant of the messaging-activemq subsystem modules.

 * Fixing the test suite
 * Tranforming messaging subsystem modules

Jira: https://issues.redhat.com/browse/WFLY-15547

### CODE REVIEW

File: messaging-activemq/subsystem/src/test/java/org/wildfly/extension/messaging/activemq/MessagingDependencies.java [hunk 1]
```diff
@@ -34,46 +34,51 @@
...
         Map<ModelTestControllerVersion, String[]> map = new HashMap<>();
-        map.put(ModelTestControllerVersion.EAP_7_3_0, new String[] {
-                "org.apache.activemq:artemis-commons:2.9.0.redhat-00005",
-                "org.apache.activemq:artemis-journal:2.9.0.redhat-00005",
-                "org.apache.activemq:artemis-server:2.9.0.redhat-00005",
-                "org.apache.activemq:artemis-jms-server:2.9.0.redhat-00005",
-                "org.apache.activemq:artemis-core-client:2.9.0.redhat-00005",
-                "org.apache.activemq:artemis-jms-client:2.9.0.redhat-00005",
-                "org.apache.activemq:artemis-ra:2.9.0.redhat-00005",
+        /*if (Boolean.getBoolean("ee9")) {*/
```
Comment (bstansberry) [inline messaging-activemq/subsystem/src/test/java/org/wildfly/extension/messaging/activemq/MessagingDependencies.java:40]: Please remove.

**Review by rhusar — CHANGES_REQUESTED** [review]
@ehsavoie BTW there are failures on the CI around:

```
&amp#27;05:59:21,613 ERROR [org.jboss.as.controller.management-operation] (management-handler-thread - 2) WFLYCTL0013: Operation ("list-all-subscriptions") failed - address: ([
    ("subsystem" => "messaging-activemq"),
    ("server" => "default"),
    ("jms-topic" => "JMSTopicManagementTestCase1645621160186")
]): java.lang.NullPointerException: Value in JsonObjects name/value pair cannot be null
  at org.glassfish.jakarta.json@1.1.6//org.glassfish.json.JsonObjectBuilderImpl.validateValue(JsonObjectBuilderImpl.java:198)
  ```

File: ee-9/source-transform/messaging-activemq/injection/pom.xml [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        <artifactId>wildfly-ee-9-source-transform-parent</artifactId>
+        <!--
+        Maintain separation between the artifact id and the version to help prevent
+        merge conflicts between commits changing the GA and those changing the V.
+        -->
+        <version>27.0.0.Beta1-SNAPSHOT</version>
+        <relativePath>../../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>wildfly-messaging-activemq-injection-jakarta</artifactId>
```
Comment (bstansberry) [inline ee-9/source-transform/messaging-activemq/injection/pom.xml:40]: To have this artifact used at runtime you need to update ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/wildfly/extension/messaging-activemq/main/module.xml to use a maven filter expression.

**Review by bstansberry — CHANGES_REQUESTED** [review]
Hi @ehsavoie I don't know if the comments I make below will help with the CI failures.

File: ee-9/feature-pack/pom.xml [hunk 3]
```diff
@@ -475,7 +476,7 @@
                             </task-properties>
                             <jakarta-transform>true</jakarta-transform>
                             <!--<jakarta-transform-configs-dir>EXTERNAL_FOLDER</jakarta-transform-configs-dir>-->
-                            <!--<jakarta-transform-verbose>true</jakarta-transform-verbose>-->
+                            <jakarta-transform-verbose>true</jakarta-transform-verbose>
```
Comment (bstansberry) [inline ee-9/feature-pack/pom.xml:479]: Please comment this out again. :)

File: ee-9/feature-pack/pom.xml [hunk 4]
```diff
@@ -534,6 +535,10 @@
                                 <exclude>org.wildfly:wildfly-clustering-infinispan-client\z</exclude>
                                 <exclude>org.wildfly:wildfly-clustering-infinispan-extension\z</exclude>
                                 <exclude>org.wildfly:wildfly-clustering-web-infinispan\z</exclude>
+                                <!-- Don't transform Messaging modules -->
+                                <exclude>org.hornetq:.\z</exclude>
+                                <exclude>org.wildfly:wildfly-messaging-activemq-subsystem\z</exclude>
+                                <exclude>org.wildfly:wildfly-messaging-activemq-subsystem\z</exclude>
```
Comment (bstansberry) [inline ee-9/feature-pack/pom.xml:541]: These last two are the same, plus I don't think they'd be getting transformed anyway as you exclude them at L751 below.

**Review by bstansberry — CHANGES_REQUESTED** [review]
@ehsavoie A couple more changes...

File: ee-9/feature-pack/pom.xml (hunk 3 above)
Comment (bstansberry) [reply ee-9/feature-pack/pom.xml]: Good idea to use false.  Once this is in we should change it to use a property so people can just set it with -D.

Comment (ehsavoie) [reply ee-9/feature-pack/pom.xml]: damned I thought i just did it ;)

File: ee-9/feature-pack/pom.xml [hunk 5]
```diff
@@ -534,6 +535,10 @@
                                 <exclude>org.wildfly:wildfly-clustering-infinispan-client\z</exclude>
                                 <exclude>org.wildfly:wildfly-clustering-infinispan-extension\z</exclude>
                                 <exclude>org.wildfly:wildfly-clustering-web-infinispan\z</exclude>
+                                <!-- Don't transform Messaging modules -->
```
Comment (bstansberry) [inline ee-9/feature-pack/pom.xml:538]: A more specific message explaining why is what I'm looking for in these comments, e.g.

'Used for legacy JMS compatibility'

File: ee-9/feature-pack/pom.xml [hunk 6]
```diff
@@ -534,6 +535,10 @@
...
+                                <exclude>org.hornetq:.\z</exclude>
```
Comment (bstansberry) [inline ee-9/feature-pack/pom.xml:539]: I still see:

[INFO] EE9: transformed hornetq-jms-client-2.4.9.Final.jar

File: ee-9/feature-pack/pom.xml [hunk 7]
```diff
@@ -534,6 +535,10 @@
...
+                                <exclude>org.wildfly:wildfly-messaging-activemq-injection\z</exclude>
```
Comment (bstansberry) [inline ee-9/feature-pack/pom.xml:540]: This is not needed; see my comment on L753

File: ee-9/feature-pack/pom.xml [hunk 8]
```diff
@@ -534,6 +535,10 @@
...
+                                <exclude>org.wildfly:wildfly-messaging-activemq-subsystem\z</exclude>
```
Comment (bstansberry) [inline ee-9/feature-pack/pom.xml:541]: This should be wildfly-messaging-activemq-subsystem-jakarta with a comment

'Uses EE 8 JSON-P internally' 

(that's what the verbose logging says is the reason it transforms the -jakarta artifact

rg.wildfly:wildfly-messaging-activemq-subsystem is not needed; see my comment at L757

File: ee-9/feature-pack/pom.xml [hunk 9]
```diff
@@ -743,6 +748,14 @@
                     <groupId>${project.groupId}</groupId>
                     <artifactId>wildfly-mail</artifactId>
                 </exclusion>
+                <exclusion>
+                    <groupId>${project.groupId}</groupId>
+                    <artifactId>wildfly-messaging-injection</artifactId>
```
Comment (bstansberry) [inline ee-9/feature-pack/pom.xml:753]: This should be 'wildfly-messaging-activemq-injection'

The missing 'activemq' is why you still needed to exclude this.

File: ee-9/feature-pack/pom.xml [hunk 10]
```diff
@@ -743,6 +748,14 @@
...
+                </exclusion>
+                <exclusion>
+                    <groupId>${project.groupId}</groupId>
+                    <artifactId>wildfly-messaging-subsystem</artifactId>
```
Comment (bstansberry) [inline ee-9/feature-pack/pom.xml:757]: This should be 'wildfly-messaging-activemq-subsystem'

The missing 'activemq' is why you still needed to exclude this in the galleon plugin config.

**Review by bstansberry — APPROVED** [review]
I'm still seeing this:

[INFO] EE9: transformed wildfly-messaging-activemq-subsystem-jakarta-27.0.0.Beta1-SNAPSHOT.jar

I assume because of the use of javax.json.

But I'll merge this and if needed can follow up excluding that.

**Pull request merged**